### PR TITLE
Make sure WebRTC preferences are synced *every* call

### PR DESCRIPTION
**Pull request by michaelkirk**
This slows the UI, but only for people who have locally opted into
WebRTC calls, and the alternative is that users are likely to have stale
settings the first time a pair of people opt-in.

// FREEBIE

### CODE REVIEW

**Comment by michaelkirk** [comment]
PTAL @charlesmchen

File: Signal/src/view controllers/MessagesViewController.m [hunk 1]
```diff
@@ -626,32 +627,62 @@ - (void)showFingerprintWithTheirIdentityKey:(NSData *)theirIdentityKey theirSign
...
+                (remoteWantsWebRTC ? @"YES" : @"NO"));
+            if (localWantsWebRTC && remoteWantsWebRTC) {
+                [self placeWebRTCCall];
+            } else {
+                [self placeRedphoneCall];
+            }
+        }
+        failure:^(NSError *_Nonnull error) {
+            DDLogWarn(@"%@ looking up call recipient: %@ failed with error: %@", self.tag, self.thread, error);
+            SignalAlertView(NSLocalizedString(@"UNABLE_TO_PLACE_CALL", @"Alert Title"), error.localizedDescription);
```
Comment (charlesmchen) [inline Signal/src/view controllers/MessagesViewController.m]: I don't think we should ever surface error descriptions to the user. I suggest: "Could not reach service." (indicating rightfully that user should check their connectivity).

File: Signal/src/view controllers/MessagesViewController.m [hunk 2]
```diff
@@ -626,32 +627,62 @@ - (void)showFingerprintWithTheirIdentityKey:(NSData *)theirIdentityKey theirSign
...
+        }];
+}
+
+- (void)placeRedphoneCall
+{
+    DDLogInfo(@"%@ Placing redphone call to: %@", self.tag, self.thread);
```
Comment (charlesmchen) [inline Signal/src/view controllers/MessagesViewController.m]: Should we wrap this logging with a `#ifdef DEBUG`?  The log files are unencrypted.

File: Signal/src/view controllers/MessagesViewController.m [hunk 3]
```diff
@@ -626,32 +627,62 @@ - (void)showFingerprintWithTheirIdentityKey:(NSData *)theirIdentityKey theirSign
...
+
+    OWSAssert(number != nil);
+    OWSAssert(contact != nil);
+
+    [Environment.phoneManager initiateOutgoingCallToContact:contact atRemoteNumber:number];
+}
+
+- (void)placeWebRTCCall
+{
+    DDLogInfo(@"%@ Placing WebRTC call to: %@", self.tag, self.thread);
```
Comment (charlesmchen) [inline Signal/src/view controllers/MessagesViewController.m]: Ditto

File: Signal/src/view controllers/MessagesViewController.m (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/view controllers/MessagesViewController.m]: We're not surfacing error descriptions, we're surfacing localizedErrorDescriptions. Arguably it should be localizedRecoverySuggestion, but we're not actively using it.

Network failures can happen for lots of reasons. e.g. offline, or the account were unregistered

In the case of disconnected network, the error reads: "Signal was unable to connect to the internet. Please try from another WiFi network or use mobile data."

File: Signal/src/view controllers/MessagesViewController.m (hunk 2 above)
Comment (michaelkirk) [inline Signal/src/view controllers/MessagesViewController.m]: I think a logging a call is valuable diagnostics. We rely on the scrubbing log formatter to remove phone numbers in non-debug.

**Comment by michaelkirk** [comment]
In https://github.com/WhisperSystems/Signal-iOS/commit/fff061ff3fb664f6e4da06c391580df37ebd9978

**Pull request closed without merging**